### PR TITLE
Update Zephyr RTOS port for Zephyr 1.10 release

### PR DESCRIPTION
**Pull request by pfalcon**
Zephyr 1.10 has switched to CMake-based build system (kind of like JerryScript, but the best way to think about it is of course as "different CMake-based build system").

Besides updating for new build system, the patch also removes some old stuff and updates for other changes in Zephyr.

### CODE REVIEW

**Comment by pfalcon** [comment]
Note that this shouldn't be merged before the official 1.10 release (ETA is beginning of December).

**Comment by pfalcon** [comment]
@SebastianBoe, @grgustaf, @jimmy-huang, @poussa : FYI.

File: targets/zephyr/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+zephyr_get_compile_definitions_as_string(definitions)
+zephyr_get_compile_options_as_string(options)
+
+add_custom_target(
+  outputexports
+  COMMAND echo CC="${CMAKE_C_COMPILER}"
+  COMMAND echo Z_CFLAGS=${system_includes}${includes}${definitions}${options}
+  COMMAND echo NOSTDINC_FLAGS=${system_includes}
+  COMMAND echo ZEPHYRINCLUDE=${includes}
+  COMMAND echo KBUILD_CFLAGS=${definitions}${options}
```
Comment (SebastianBoe) [inline targets/zephyr/CMakeLists.txt]: Glad to see that this worked :)

**Comment by pfalcon** [comment]
Update: Makefile.zephyr was updated, CI now passes.

**Comment by pfalcon** [comment]
@akosthekiss : Can you please review Makefile.travis changes (as updating your https://github.com/jerryscript-project/jerryscript/pull/2102) ?

**Comment by pfalcon** [comment]
Updated for latest changes in Zephyr build system.

**Comment by pfalcon** [comment]
Zephyr 1.10 has been released, so this is ready for merge now.

@akosthekiss, @yichoi, @LaszloLango, @dbatyai : Can you please review?

File: targets/zephyr/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,31 @@
+include($ENV{ZEPHYR_BASE}/cmake/app/boilerplate.cmake NO_POLICY_SCOPE)
```
Comment (LaszloLango) [inline targets/zephyr/CMakeLists.txt]: Please add the license header.

File: targets/zephyr/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+add_library(jerry-core STATIC IMPORTED)
+add_library(jerry-ext STATIC IMPORTED)
+set_target_properties(jerry-core PROPERTIES IMPORTED_LOCATION ${CMAKE_CURRENT_BINARY_DIR}/../obj/lib/libjerry-core.a)
+set_target_properties(jerry-core PROPERTIES INTERFACE_INCLUDE_DIRECTORIES ${CMAKE_CURRENT_SOURCE_DIR}/../../jerry-core/include)
+set_target_properties(jerry-ext PROPERTIES IMPORTED_LOCATION ${CMAKE_CURRENT_BINARY_DIR}/../obj/lib/libjerry-ext.a)
+set_target_properties(jerry-ext PROPERTIES INTERFACE_INCLUDE_DIRECTORIES ${CMAKE_CURRENT_SOURCE_DIR}/../../jerry-ext/include)
+target_link_libraries(app jerry-core jerry-ext)
+
+
+
```
Comment (LaszloLango) [inline targets/zephyr/CMakeLists.txt]: Too many empty lines.

File: targets/zephyr/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+project(NONE)
+
+target_sources(app PRIVATE src/main-zephyr.c src/jerry-port.c src/getline-zephyr.c)
+
+add_library(jerry-core STATIC IMPORTED)
+add_library(jerry-ext STATIC IMPORTED)
+set_target_properties(jerry-core PROPERTIES IMPORTED_LOCATION ${CMAKE_CURRENT_BINARY_DIR}/../obj/lib/libjerry-core.a)
+set_target_properties(jerry-core PROPERTIES INTERFACE_INCLUDE_DIRECTORIES ${CMAKE_CURRENT_SOURCE_DIR}/../../jerry-core/include)
+set_target_properties(jerry-ext PROPERTIES IMPORTED_LOCATION ${CMAKE_CURRENT_BINARY_DIR}/../obj/lib/libjerry-ext.a)
+set_target_properties(jerry-ext PROPERTIES INTERFACE_INCLUDE_DIRECTORIES ${CMAKE_CURRENT_SOURCE_DIR}/../../jerry-ext/include)
```
Comment (LaszloLango) [inline targets/zephyr/CMakeLists.txt]: These `set_target_properties` lines are quite long. What about wrapping them?

For example:
```
set_target_properties(jerry-ext PROPERTIES INTERFACE_INCLUDE_DIRECTORIES
                      ${CMAKE_CURRENT_SOURCE_DIR}/../../jerry-ext/include)
```

File: targets/zephyr/Makefile.travis [hunk 5]
```diff
@@ -27,12 +27,15 @@ install-apt-get-deps:
...
+	wget https://github.com/zephyrproject-rtos/meta-zephyr-sdk/releases/download/0.9.2/zephyr-sdk-0.9.2-setup.run -O ../zephyr-sdk-0.9.2-setup.run
+	sh ../zephyr-sdk-0.9.2-setup.run -- -y -d $(CURDIR)/../zephyr-sdk-0.9.2
+	wget https://cmake.org/files/v3.8/cmake-3.8.2-Linux-x86_64.sh
+	sudo sh cmake-3.8.2-Linux-x86_64.sh --prefix=/usr/local --exclude-subdir
+	cmake --version
 
 # Fetch Zephyr Project repository and install python dependencies.
 install-zephyr: install-apt-get-deps
-	git clone https://github.com/zephyrproject-rtos/zephyr.git ../zephyr -b zephyr-v1.9.1
+	git clone https://github.com/zephyrproject-rtos/zephyr.git ../zephyr -b v1.10-branch
```
Comment (LaszloLango) [inline targets/zephyr/Makefile.travis]: Will this branch live forever? IMHO it will be better to use the release tag and not the branch.

**Comment by LaszloLango** [comment]
Nice patch, I have just a few minor comments.

File: targets/zephyr/CMakeLists.txt (hunk 2 above)
Comment (pfalcon) [reply targets/zephyr/CMakeLists.txt]: Done

File: targets/zephyr/CMakeLists.txt (hunk 4 above)
Comment (pfalcon) [reply targets/zephyr/CMakeLists.txt]: Done

File: targets/zephyr/CMakeLists.txt (hunk 3 above)
Comment (pfalcon) [reply targets/zephyr/CMakeLists.txt]: Fixed

File: targets/zephyr/Makefile.travis (hunk 5 above)
Comment (pfalcon) [reply targets/zephyr/Makefile.travis]: This branch would transition thru 1.10.x point releases, and the idea was to check that JrS remains buildable with them, but as you wish, switching to the tag.

**Comment by pfalcon** [comment]
> LGTM, but please squash the commits with an appropriate commit message.

Thanks! Originally, these commits were intended to be separate. If your request them to be squashed, let me do that in the separate PR: https://github.com/jerryscript-project/jerryscript/pull/2155

**Pull request closed without merging**